### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by gapiyka**

### CODE REVIEW

File: js/game.js [hunk 1]
```diff
@@ -0,0 +1,596 @@
...
+            modelCut = 0;
+        }
+        else if (!this.lookRight && this.lookForward) {
+            modelCut = 40;
+        }
+        else if (this.lookRight && !this.lookForward) {
+            modelCut = 80;
+        }
+        else if (!this.lookRight && !this.lookForward) {
+            modelCut = 120;
```
Comment (tshemsedinov) [inline js/game.js:56]: Magic numbers, use named constants or collection

File: js/game.js [hunk 2]
```diff
@@ -0,0 +1,596 @@
...
+        }
+        ctx.drawImage(this.image, modelCut, 0, this.width, this.height, this.position.x - this.width / 2, this.position.y, 80, 40);
```
Comment (tshemsedinov) [inline js/game.js:58]: Long line, decompose, use intermediate variables

File: js/game.js [hunk 3]
```diff
@@ -0,0 +1,596 @@
...
+        if (this.hp > 50) ctx.fillStyle = "green";
+        else if (this.hp < 25) ctx.fillStyle = "red";
+        else ctx.fillStyle = "yellow";
+        ctx.fillRect(this.position.x, this.position.y - 15, this.hp / 2.5, 10);
```
Comment (tshemsedinov) [inline js/game.js:62]: Magic numbers everywhere
AmbitiousPracticalJabiru-small

File: js/game.js [hunk 4]
```diff
@@ -0,0 +1,596 @@
...
+                    break;
+                case 68://D
+                    player.moveRight();
+                    break;
+                case 83://S
+                    player.moveDown();
+                    break;
+                case 32://space
+                    if (!IsGameStarted) StartGame();
+                    break;
```
Comment (tshemsedinov) [inline js/game.js:110]: Use collection instead, see lectures about antipatterns and refactoring

File: js/game.js [hunk 5]
```diff
@@ -0,0 +1,596 @@
...
+                    if (player.speed.x < 0) player.stopX();
+                    break;
+                case 87:
+                    if (player.speed.y < 0) player.stopY();
+                    break;
+                case 68:
+                    if (player.speed.x > 0) player.stopX();
+                    break;
+                case 83:
+                    if (player.speed.y > 0) player.stopY();
```
Comment (tshemsedinov) [inline js/game.js:137]: Code duplication

File: js/game.js [hunk 6]
```diff
@@ -0,0 +1,596 @@
...
+let topTimeScore = 0;
+let startGameTime;
+let IsGameStarted = false;
+let IsNightMode = false;
+let nightCounter;
+let staticObjects = [];
+let fruit;
+let vulture;
+let secDelta;
+let ScoreRating = [];
```
Comment (tshemsedinov) [inline js/game.js:350]: Holly crap, mutable global scope
WImtWKc

**Review by tshemsedinov — CHANGES_REQUESTED** [review]
- Refactor code duplication
- Use constants instead of magic numbers
- Use collections instead of `swith/case`
- Add eslint and prettier

**Comment by gapiyka** [comment]
Fixed all except `mutable global scope` in next commit *main

image

File: js/game.js [hunk 7]
```diff
@@ -0,0 +1,596 @@
+'use strict';
+
+
+//CLASES:
+class Player {
+    constructor(gameWidth, gameHeight, image) {
+        this.width = 40;
+        this.height = 40;
```
Comment (tshemsedinov) [inline js/game.js:8]: Magic numbers

File: js/game.js [hunk 8]
```diff
@@ -0,0 +1,596 @@
...
+        this.image = image;
+        this.maxSpeed = 3;
+        this.speed = { x: 0, y: 0 };
+        this.position = {
+            x: gameWidth / 2 - this.width / 2,
+            y: gameHeight - this.height - 10
+        };
```
Comment (tshemsedinov) [inline js/game.js:15]: Magic everywhere

File: js/game.js [hunk 9]
```diff
@@ -0,0 +1,596 @@
...
+        this.colision = this.position;
+        this.lookForward = true;
+        this.lookRight = true;
```
Comment (tshemsedinov) [inline js/game.js:18]:
```js
this.lookDirection = 'forward';
this.lookAngle = 'right';
```

**Pull request closed without merging**